src/Middleware/HttpLogging/src/HttpLoggingMiddleware.cs

```diff
@@ -131,10 +131,25 @@ private async Task InvokeInternal(HttpContext context)
         ResponseBufferingStream? responseBufferingStream = null;
         IHttpResponseBodyFeature? originalBodyFeature = null;
 
+        UpgradeFeatureLoggingDecorator? loggableUpgradeFeature = null;
+        IHttpUpgradeFeature? originalUpgradeFeature = null;
+
         try
         {
             var response = context.Response;
 
+            if (options.LoggingFields.HasFlag(HttpLoggingFields.ResponsePropertiesAndHeaders))
```

Comment:
I made this change according to the specs, but I was wondering whether or not it's really important that we do this check here to avoid wrapping the upgrade feature in case we are not interested in logging the response headers or status code. The reasoning is that the overhead seems minimal, the log will be filtered anyway when the call to the logging system happen and it would simplify the code a bit.